### ISSUE TITLE
Clarify padded A env-mat normalization semantics across GPU, CPU, and dpmodel

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#5584]
While reviewing #5582, I noticed a subtle mismatch in how padded neighbor slots are normalized in the A environment matrix. This issue is for documenting and confirming the intended semantics, rather than claiming a confirmed downstream `se_atten` regression.

## Observation

For an A descriptor neighbor slot with `nlist[ii] < 0`, the raw environment-matrix value is conceptually a zero descriptor. If normalization is applied uniformly, the normalized value should be

```text
(0 - avg[type][slot_component]) / std[type][slot_component]
```

for all 4 A components of that padded slot.

In the older TF GPU kernel, real neighbor slots normalized all 4 components, but the padded-slot branch only touched the first component:

https://github.com/deepmodeling/deepmd-kit/blob/1a8586b8925ac6e2e9953b0f61dcfd8b96c748b5/source/lib/src/gpu/prod_env_mat.cu#L513-L520

```cpp
for (int ii = 0; ii < 4; ii++) {
  row_descript[idx_value + ii] =
      (dd[ii] - avg[type[i_idx_nall] * ndescrpt + idx_value + ii]) /
      std[type[i_idx_nall] * ndescrpt + idx_value + ii];
}
} else {
  row_descript[idx_value] -= avg[type[i_idx_nall] * ndescrpt + idx_value] /
                           std[type[i_idx_nall] * ndescrpt + idx_value];
}
```

So for padded slots, components 1..3 were not normalized or assigned by this branch. This is observable in low-level env-mat output if `avg` is nonzero.

In #5582, I changed the GPU branch to apply the same normalized-zero rule to all 4 components:

https://github.com/deepmodeling/deepmd-kit/blob/048376baa460d0082c3fbfef1a31c244d3f1a727/source/lib/src/gpu/prod_env_mat.cu#L513-L523

```cpp
} else {
  for (int ii = 0; ii < 4; ii++) {
    row_descript[idx_value + ii] -=
        avg[type[i_idx_nall] * ndescrpt + idx_value + ii] /
        std[type[i_idx_nall] * ndescrpt + idx_value + ii];
  }
}
```

## Comparison with `dpmodel`

`dpmodel` appears to follow the uniform-normalization semantics.

First, `_make_env_mat` masks absent neighbors, so the raw env-mat for padded slots is zero:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/utils/env_mat.py#L88-L104

```python
weight = weight * xp.astype(xp.expand_dims(mask, axis=-1), weight.dtype)
if radial_only:
    env_mat = t0 * weight
else:
    env_mat = xp.concat([t0, t1], axis=-1) * weight
```

Then `EnvMat.call` subtracts `davg` and divides by `dstd` for the whole `em` tensor:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/utils/env_mat.py#L158-L165

```python
if davg is not None:
    em -= xp.reshape(xp.take(davg, xp.reshape(atype, (-1,)), axis=0), em.shape)
if dstd is not None:
    em /= xp.reshape(xp.take(dstd, xp.reshape(atype, (-1,)), axis=0), em.shape)
```

Therefore, for padded A slots, `dpmodel` produces `-davg/dstd` for all 4 components, matching the #5582 GPU change and not the older TF GPU padded-slot branch.

## Downstream masking in `se_atten` / DPA1

This mismatch may not affect the usual final descriptor in the standard attention path, because padded neighbors are also masked downstream.

For TF `se_atten`, the op returns `nmask`, which is converted to `negative_mask`:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/tf/descriptor/se_atten.py#L639-L644

```python
self.nmask = tf.cast(
    tf.reshape(self.nmask, [-1, 1, self.sel_all_a[0]]),
    self.filter_precision,
)
self.negative_mask = -(2 << 32) * (1.0 - self.nmask)
```

The attention weights are masked for absent neighbors:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/tf/descriptor/se_atten.py#L968-L978

```python
attn *= self.nmask
attn += self.negative_mask
attn = tf.nn.softmax(attn, axis=-1)
...
attn *= tf.reshape(self.nmask, [-1, attn.shape[-1], 1])
```

The final contraction then uses the attention result:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/tf/descriptor/se_atten.py#L1309-L1313

```python
return tf.matmul(
    tf.reshape(inputs_i, [natom, shape_i[1] // 4, 4]),
    xyz_scatter_att,
    transpose_a=True,
)
```

For `dpmodel` DPA1, the env mat is normalized first:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/descriptor/dpa1.py#L1096-L1102

and then `nlist_mask = nlist != -1` is passed into attention:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/descriptor/dpa1.py#L1117-L1122

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/descriptor/dpa1.py#L1220-L1222

and absent neighbors are masked in attention:

https://github.com/deepmodeling/deepmd-kit/blob/b52c359ec6b6f6f34998e00f774c35d253cf09e5/deepmd/dpmodel/descriptor/dpa1.py#L1663-L1670

This suggests that, for the usual attention path, padded env-mat values should not contribute physically even when they differ. Also, if `davg` is zero, this difference is numerically invisible.

## Questions

1. Should env-mat operators define padded A slots as normalized zero descriptors, i.e. `-avg/std` for every component, even if most consumers also carry an explicit mask?
2. Or should padded slots be forced to zero after normalization, making consumers less sensitive to whether they mask correctly?
3. Are there non-attention paths, `attn_layer == 0` paths, compression/tabulation paths, or standalone op users where padded env-mat values can affect results when `davg` is nonzero?
4. Should we add explicit tests asserting CPU/GPU/`dpmodel` consistency for padded slots with nonzero `avg`?

My current reading is that the #5582 GPU change makes the low-level TF GPU env-mat behavior consistent with `dpmodel` and with uniform normalization. I would like to confirm that this is the intended public semantics before relying on it more broadly.
